### ISSUE TITLE
Docker container pulled with singularity 2.5.2 works with 3.2.0 exec, but fails if pulled with 3.2.0

### ISSUE THREAD

**Issue by keiranmraine** [apptainer/singularity#3594]
3.2.0 continues to have problems with pulled docker containers, but works fine with containers pulled with 2.5.2.

Same issue as in #3331

I can get the expected behaviour by adding `--cleanenv` but I would expect absence of this to result in host-os values being appended, not completely overriding those defined in the container.

## Version of Singularity:

3.2.0

### Expected behavior

Env variables set in dockerfile should be honoured.

Expect to find PERL5LIB of `PERL5LIB=/opt/wtsi-cgp/lib/perl5`.

```
$ /software/singularity-2.5.2/bin/singularity pull docker://quay.io/wtsicgp/cgppindel:hotfix_v3.2.1
$ /software/singularity-v3.2.0/bin/singularity exec --workdir $PWD --home $PWD:/home --pwd /home cgppindel-hotfix_v3.2.1.simg env | grep PERL5LIB
PERL5LIB=/opt/wtsi-cgp/lib/perl5
```

At worst the "host-os" variables could be appended giving:

```
PERL5LIB=/opt/wtsi-cgp/lib/perl5:/nfs/users/me/lib/perl5
```

### Actual behavior

Env variables from host OS are taking precedence over those explicitly defined in the container.

```
$ /software/singularity-3.2.0/bin/singularity pull docker://quay.io/wtsicgp/cgppindel:hotfix_v3.2.1
$ /software/singularity-3.2.0/bin/singularity exec --workdir $PWD --home $PWD:/home --pwd /home cgppindel-hotfix_v3.2.1.simg env | grep PERL5LIB
PERL5LIB=/nfs/users/me/lib/perl5
```

Libs required for the container to function are lost as the "host-os" inherited env completely overrides.

### Steps to reproduce behavior

As above.

**Comment by AdamSimpson** [comment]
Also discussed in #3510

**Comment by dtrudg** [comment]
Pulling with singularity `3.5.2` now gives:

```
dave@ythel~> singularity exec --workdir $PWD --home $PWD:/home --pwd /home cgppindel_hotfix_v3.2.1.sif env | grep PERL5LIB
PERL5LIB=/opt/wtsi-cgp/lib/perl5
```

**Comment by dtrudg** [comment]
Not true.... I hadn't exported the host `PERL5LIB` as I thought I had... then it is...

```
dave@ythel~> export PERL5LIB="/nfs"
dave@ythel~> singularity exec --workdir $PWD --home $PWD:/home --pwd /home cgppindel_hotfix_v3.2.1.sif env | grep PERL5LIB
PERL5LIB=/nfs
```

**Comment by cclerget** [comment]
Fixed by #5028
